core/src/banking_stage/consumer.rs

```diff
@@ -256,23 +271,39 @@ impl Consumer {
                 // following are retryable errors
                 Err(TransactionError::AccountInUse) => {
                     error_counters.account_in_use += 1;
-                    Some(index)
+                    // locking failure due to vote conflict or jito - immediately retry.
+                    Some(RetryableIndex {
+                        index,
+                        immediately_retryable: true,
+                    })
                 }
                 Err(TransactionError::WouldExceedMaxBlockCostLimit) => {
                     error_counters.would_exceed_max_block_cost_limit += 1;
-                    Some(index)
+                    Some(RetryableIndex {
+                        index,
+                        immediately_retryable: false,
```

Comment:
Just think out loud, it's possible that a transaction rejected due to a small CU overage could still fit within the current slot after other in-flight transactions release unused CUs. If the `WouldExceed` errors included overage details, we might be able to give transactions the option to retry immediately.

Reply:
I think the solution here is separate, in that we need to abandon the reserve-commit approach to CU tracking.

Reply:
Even with the reserve-commit regime, it’s possible to return an error like “cannot reserve at this time, as the transaction requires 1 CU more than currently available.” In such cases, the transaction could be retried immediately. But I agree this would be a separate topic.